### ISSUE TITLE
anyOf produces invalid output

### ISSUE THREAD

**Issue by cozmo** [drwpow/openapi-typescript#305]
Given the following spec (which is valid/correct per https://www.jsonschemavalidator.net)

```json
{
  "openapi": "3.0",
  "components": {
    "schemas": {
      "scalar_map": {
        "type": "object",
        "additionalProperties": {
          "oneOf": [
            {
              "type": "string"
            },
            {
              "type": "number"
            },
            {
              "type": "boolean"
            }
          ]
        }
      }
    }
  }
}
```

swagger-to-ts outputs

```ts
/**
 * This file was auto-generated by swagger-to-ts.
 * Do not make direct changes to the file.
 */

export interface components {
  schemas: {
    scalar_map: { [key: string]: oneOf };
  };
}
```

which is malformed. The test case added in #201 works correctly, but it seems like maybe that's a unique/happy path? Seems like `anyOf` support still has some issues. 

If you point me in the right direction I'd be happy to take a stab at fixing this, but at first pass not really sure where to start.

**Comment by vultix** [comment]
I'm running into the same issue, but I'm using a `$ref` inside of `additionalProperties`.  

This outputs `[key: string]: ref` which is malformed.

In addition, the description inside `additionalProperties` isn't being respected.

**Comment by eysi09** [comment]
I'm running into both of the issues described above.

**Comment by drwpow** [comment]
Fixed in #442!
